app/code/Oleksandrr/Chat/Controller/Message/Save.php

```diff
@@ -29,50 +30,81 @@ class Save extends \Magento\Framework\App\Action\Action implements
      */
     private $messageResource;
 
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    private $userSession;
+
+    /**
+     * @var \Magento\Framework\Data\Form\FormKey\Validator
+     */
+    private $formKeyValidator;
+
     /**
      * Save constructor.
      * @param \Magento\Framework\App\Action\Context $context
      * @param \Oleksandrr\Chat\Model\MessageFactory $messageFactory
      * @param \Magento\Store\Model\StoreManagerInterface $storeManager
      * @param \Psr\Log\LoggerInterface $logger
-     * @param \Oleksandrr\Chat\Model\ResourceModel\Message $messageResource
+     * @param \Oleksandrr\Chat\Model\ResourceModel\Message $messageResource,
+     * @param \Magento\Customer\Model\Session $userSession
      */
     public function __construct(
         \Magento\Framework\App\Action\Context $context,
         \Oleksandrr\Chat\Model\MessageFactory $messageFactory,
         \Magento\Store\Model\StoreManagerInterface $storeManager,
         \Psr\Log\LoggerInterface $logger,
-        \Oleksandrr\Chat\Model\ResourceModel\Message $messageResource
+        \Oleksandrr\Chat\Model\ResourceModel\Message $messageResource,
+        \Magento\Customer\Model\Session $userSession,
+        \Magento\Framework\Data\Form\FormKey\Validator $formKeyValidator
     ) {
         parent::__construct($context);
         $this->messageFactory = $messageFactory;
         $this->storeManager = $storeManager;
         $this->logger = $logger;
         $this->messageResource = $messageResource;
+        $this->userSession = $userSession;
+        $this->formKeyValidator = $formKeyValidator;
     }
 
     /**
      * @inheritDoc
      */
     public function execute()
     {
+        /** @var \Magento\Framework\App\Request\Http $request */
+        $request = $this->getRequest();
+        $adminMessage = '';
+
         try {
-            /** @var \Magento\Framework\App\Request\Http $request */
-            $request = $this->getRequest();
+            if (!$this->formKeyValidator->validate($request)) {
+                throw new LocalizedException(__('Validation Failed'));
+            }
+
+            $userId = (int) $this->userSession->getId();
             $websiteId = (int) $this->storeManager->getWebsite()->getId();
 
             /** @var \Oleksandrr\Chat\Model\Message $message */
             $message = $this->messageFactory->create();
+            $date = new \DateTime();
+
+            if (!$hash = $this->userSession->getChatHash()) {
+                $hash = md5($date->getTimestamp() . $request->getParam('user_message'));
+                $this->userSession->setChatHash($hash);
+            }
 
-            $message->setAuthorId(1)
+            $message
+                ->setAuthorId($userId ?: 0)
                 ->setAuthorType(1)
                 ->setWebsiteId($websiteId)
                 ->setMessage($request->getParam('user_message') ?: '')
```

Comment:
@pandaUbuntu 
1) This code has better equivalent because the method `getParam` takes two arguments: `$request->getParam('user_message', '')`
2) Empty message is a problem and this should trigger error.